gpkit/small_classes.py

```diff
@@ -247,7 +247,14 @@ def __add__(self, other):
         elif isinstance(other, dict):
             sums = self.copy()
             for key, value in other.items():
-                sums[key] = value + sums.get(key, 0)
+                if key in sums:
+                    svalue = sums[key]
+                    if value == -svalue:
+                        del sums[key]
```

Comment:
remove zeros created during addition! (but not if they were already present in self or other)